Source/Core/VideoBackends/Vulkan/VulkanLoader.cpp

```diff
@@ -13,94 +13,95 @@
 
 #if defined(VK_USE_PLATFORM_WIN32_KHR)
 #include <Windows.h>
+#define LOADER_WINDOWS 1
+
 #elif defined(VK_USE_PLATFORM_XLIB_KHR) || defined(VK_USE_PLATFORM_XCB_KHR) ||                     \
     defined(VK_USE_PLATFORM_ANDROID_KHR) || defined(USE_HEADLESS)
 #include <dlfcn.h>
+#define LOADER_UNIX 1
 #endif
 
-#define VULKAN_MODULE_ENTRY_POINT(name, required) PFN_##name name;
-#define VULKAN_INSTANCE_ENTRY_POINT(name, required) PFN_##name name;
-#define VULKAN_DEVICE_ENTRY_POINT(name, required) PFN_##name name;
-#include "VideoBackends/Vulkan/VulkanEntryPoints.inl"
-#undef VULKAN_DEVICE_ENTRY_POINT
-#undef VULKAN_INSTANCE_ENTRY_POINT
-#undef VULKAN_MODULE_ENTRY_POINT
+namespace
+{
+#if defined(LOADER_WINDOWS)
+using Module = HMODULE;
+const char* search_lib_names[] = {"vulkan-1.dll"};
 
-namespace Vulkan
+bool LoadModule(Module* module, const char* lib_name) noexcept
 {
-static void ResetVulkanLibraryFunctionPointers()
+  *module = LoadLibraryA(lib_name);
+
+  return module != nullptr;
```

Comment:
Should this be *module != nullptr?

Or better yet, drop the second level of indirection. As far as I'm aware, on any of the platforms we care about, the shared library handle is always a pointer.

Reply:
I believe, that's exactly what stenzek have been proposing.
Smart pointers isn't a good idea here, because we neither transfer ownership, nor "branching" it, but rather temporarily giving it away and controlling lifetime from the caller.

The reason, it uses a pointer, is to make api consistent, as UnloadModule is accepting a pointer. Also, it's a small static function, so any good compiler would inline it and generate exactly equivalent binary.

Reply:
```module != nullptr``` is exactly a bug, thanks!